packages/codegen-ui-react/lib/react-component-render-helper.ts

```diff
@@ -404,36 +394,38 @@ export function buildChildElement(prop?: ComponentPropertyValueTypes): JsxChild
   return expression && factory.createJsxExpression(undefined, expression);
 }
 
-export function buildOpeningElementAttributes(prop: ComponentPropertyValueTypes, propName: string): JsxAttribute {
+export function buildOpeningElementAttributes(prop: StudioComponentProperty, name: string): JsxAttribute {
   if (isFixedPropertyWithValue(prop)) {
-    return buildFixedAttr(prop, propName);
+    return buildFixedAttr(prop, name);
   }
   if (isBoundProperty(prop)) {
     return prop.defaultValue === undefined
-      ? buildBindingAttr(prop, propName)
-      : buildBindingAttrWithDefault(prop, propName, prop.defaultValue);
+      ? buildBindingAttr(prop, name)
+      : buildBindingAttrWithDefault(prop, name, prop.defaultValue);
   }
   if (isCollectionItemBoundProperty(prop)) {
     return prop.defaultValue === undefined
-      ? buildCollectionBindingAttr(prop, propName)
-      : buildCollectionBindingAttrWithDefault(prop, propName, prop.defaultValue);
+      ? buildCollectionBindingAttr(prop, name)
+      : buildCollectionBindingAttrWithDefault(prop, name, prop.defaultValue);
   }
   if (isConcatenatedProperty(prop)) {
-    return buildConcatAttr(prop, propName);
+    return buildConcatAttr(prop, name);
   }
   if (isConditionalProperty(prop)) {
-    return buildConditionalAttr(prop, propName);
+    return buildConditionalAttr(prop, name);
   }
-  return factory.createJsxAttribute(factory.createIdentifier(propName), undefined);
+  return factory.createJsxAttribute(factory.createIdentifier(name), undefined);
 }
 
 /* Tempory stub function to map from generic event name to React event name. Final implementation will be included in
  * amplify-ui.
  */
-function mapGenericEventToReact(genericEventBinding: string): string {
+export function mapGenericEventToReact(genericEventBinding: string): string {
   switch (genericEventBinding) {
     case 'click':
```

Comment:
I feel a bit as though the keys to this map should like the codegen-ui repo, so we can just enforce their names in the yup model.